include/benchmark/benchmark_api.h

```diff
@@ -858,6 +861,10 @@ class Fixture : public internal::Benchmark {
 #define BENCHMARK_MAIN()                   \
   int main(int argc, char** argv) {        \
     ::benchmark::Initialize(&argc, argv);  \
+    if (argc > 1) {                        \
```

Comment:
It's necessary if we want to treat unrecognized arguments as fatal errors and exit the application (see `return 1;` two lines below). If the benchmark should still be executed (which IMHO should not be the case, but other people might have a different opinion about that), we can remove the `if` and `return 1;`.

I also considered adding `exit(1);` to `ReportUnrecognizedArguments()` when `argc > 1`, but I figured it would not be obvious that the function can actually cause the application to terminate.